src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/KubernetesCredentialsStore.java

```diff
@@ -35,14 +37,17 @@ public ModelObject getContext() {
 
     @Override
     public boolean hasPermission(@NonNull Authentication authentication, @NonNull Permission permission) {
-        return Jenkins.getInstance().getACL().hasPermission(authentication, permission);
+        return CredentialsProvider.VIEW.equals(permission) &&
+               Jenkins.getInstance().getACL().hasPermission(authentication, permission);
     }
 
     @NonNull
     @Override
     public List<Credentials> getCredentials(@NonNull Domain domain) {
-        // TODO: Filter by domain - how do I do this?
-        return  provider.getCredentials(Credentials.class, Jenkins.getInstance(), ACL.SYSTEM);
+        // Only the global domain is supported
+        return Domain.global().equals(domain)
+               ? provider.getCredentials(Credentials.class, Jenkins.getInstance(), ACL.SYSTEM)
```

Comment:
Ahh I think you need to pass `jenkins.getAuthentication()` rather than `ACL.SYSTEM`

Reply:
If I do that, then [this check](https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/blob/0e4ab0963e8c9f9ed455cc159a881b63438e474d/src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/KubernetesCredentialProvider.java#L126) fails and we don't see the credentials

Reply:
Hmm.. I used [this code](https://github.com/jenkinsci/credentials-plugin/blob/845bb7946a53650da1eeb457152d4296eaeb58c5/src/main/java/com/cloudbees/plugins/credentials/SystemCredentialsProvider.java#L436-L441) as a reference for that.

Reply:
I could do a similar check to [this](https://github.com/jenkinsci/credentials-plugin/blob/845bb7946a53650da1eeb457152d4296eaeb58c5/src/main/java/com/cloudbees/plugins/credentials/SystemCredentialsProvider.java#L289) - ie:
```
    public List<Credentials> getCredentials(@NonNull Domain domain) {
        // Only the global domain is supported
        if (Domain.global().equals(domain) && Jenkins.getInstance().hasPermission(CredentialsProvider.VIEW))
            return provider.getCredentials(Credentials.class, Jenkins.getInstance(), ACL.SYSTEM);
        return Collections.emptyList();
    }
```